scripts/playerhandlesprings/playerhandlesprings.gml

```diff
@@ -29,7 +29,9 @@ function PlayerHandleSprings(){
 	if (oSpring) {
 		if (abs(angle_difference(sensor.get_angle(), oSpring.image_angle - 90)) < 60 &&
 			oSpring.image_speed == 0
-		) {	
+		) {
+			if(state.current() != "roll") state.change_to("normal");
```

Comment:
What is the reason of this?

Reply:
Because if Knuckles glide or land in spring, he shoaled walk beck

Reply:
I don't think this is a problem. But if it is, why not just check the "glider" and "ground" states directly? I think that would be better.

Reply:
Fine, I changed my mind, mb it's fine